react/Trigger.tsx

```diff
@@ -1,60 +1,90 @@
+import {
+  useResponsiveValue,
+  MaybeResponsiveInput,
+} from 'vtex.responsive-values'
 import classnames from 'classnames'
 import { useCssHandles } from 'vtex.css-handles'
 import React, { useEffect, useRef } from 'react'
 
 import {
+  usePopoverState,
   usePopoverDispatch,
   PopoverContextProvider,
 } from './components/PopoverContext'
 
 const CSS_HANDLES = ['trigger']
 
-// TODO hover trigger
-type TriggerMode = 'click' | 'hover'
+export type TriggerMode = 'click' | 'hover'
 
 interface Props {
   children: React.ReactNode
-  trigger?: TriggerMode
+  trigger?: MaybeResponsiveInput<TriggerMode>
 }
 
 export type TriggerElement = HTMLDivElement & HTMLButtonElement
 
 function Trigger(props: Props) {
-  const { children, trigger = 'click' } = props
+  const { children, trigger: triggerProp = 'click' } = props
+  const trigger = useResponsiveValue(triggerProp)
   const dispatch = usePopoverDispatch()
+  const { open } = usePopoverState()
   const containerRef = useRef<TriggerElement>(null)
   const handles = useCssHandles(CSS_HANDLES)
   const ContainerTag = trigger === 'click' ? 'button' : 'div'
 
   useEffect(() => {
-    if (dispatch) {
-      dispatch({
-        type: 'SET_CONTAINER_REF',
-        payload: { containerRef },
-      })
-    }
+    dispatch({
+      type: 'SET_CONTAINER_REF',
+      payload: { containerRef },
+    })
   }, [dispatch, containerRef])
 
+  useEffect(() => {
+    dispatch({
+      type: 'SET_TRIGGER_MODE',
+      payload: { triggerMode: trigger },
+    })
+  }, [dispatch, trigger])
+
   const handleKeydown = (e: React.KeyboardEvent<TriggerElement>) => {
-    if (e.key !== 'Enter') {
+    if (e.key !== 'Enter' || trigger !== 'click') {
       return
     }
 
-    if (dispatch) {
-      e.stopPropagation()
-      dispatch({ type: 'OPEN_POPOVER' })
-    }
+    e.stopPropagation()
+    dispatch({ type: 'OPEN_POPOVER' })
   }
 
   const handleClick = (e: React.MouseEvent<TriggerElement>) => {
+    if (trigger !== 'click') {
+      return
+    }
+    // This is needed to block some parent element to use this event
+    // to do some action (like open a Modal or change the url).
+    // If you click in the Trigger you want to open the trigger only
+    // nothing else.
     e.stopPropagation()
     e.preventDefault()
 
-    if (dispatch) {
+    if (open) {
+      dispatch({ type: 'CLOSE_POPOVER' })
+    } else {
+      dispatch({ type: 'OPEN_POPOVER' })
+    }
+  }
+
+  const handleMouseEnter = () => {
+    if (trigger === 'hover') {
```

Comment:
I think you could just not add this function handler on onMouseEnter if it's not hover